### PR TITLE
feat(CO-770): remove UC Service and Voice features

### PR DESCRIPTION
**Pull request by Polpetta**
See https://zextras.atlassian.net/browse/CO-770

### CODE REVIEW

**Comment by keshavbhatt** [comment]
@Polpetta we can drop it.
https://wiki.zimbra.com/wiki/UCIntegration

**Comment by keshavbhatt** [comment]
We can drop the whole VoiceConstants class and its usages. There are many classes that can be cleaned to remove UCService.

image

**Comment by Polpetta** [comment]
Ty @keshavbhatt, I'm gonna search and clean up the whole `VoiceConstants` stuff

**Review by keshavbhatt — CHANGES_REQUESTED** [review]
Apart from a few suggestion I made,

**We can also get rid of the following classes:**

- soap/src/main/java/com/zimbra/soap/admin/type/UCServiceInfo.java
- soap/src/main/java/com/zimbra/soap/admin/type/VoiceProviderInfo.java
- soap/src/main/java/com/zimbra/soap/admin/type/UCServiceAttribute.java
- soap/src/main/java/com/zimbra/soap/admin/message/ModifyUCService*
- soap/src/main/java/com/zimbra/soap/admin/message/CreateUCService*
- soap/src/main/java/com/zimbra/soap/admin/message/GetAllUCServices*
- soap/src/main/java/com/zimbra/soap/admin/message/RenameUCService*
- soap/src/main/java/com/zimbra/soap/admin/message/DeleteUCService*

This will also require minor cleanup in JaxbUtil and AdminConstants...

**There are still a entry related to UCService in `store-conf/conf/msgs`:**

We can get rid of them running the following command in store-conf/conf/msgs directory.

`for file in ./*; do sed -i '/listUCService/d' "$file"; done`

File: store/src/main/java/com/zimbra/cs/account/ProvUtil.java [hunk 1]
```diff
@@ -3129,49 +3065,6 @@ private void doCountAccount(String[] args) throws ServiceException {
     console.println();
   }
 
-  private void doCountObjects(String[] args) throws ServiceException {
```
[the remark this answers is not in the document]
Comment (Polpetta) [reply store/src/main/java/com/zimbra/cs/account/ProvUtil.java:3132]: I didn't know if they were related to UCService counting only

File: soap/src/main/java/com/zimbra/soap/admin/message/UpdatePresenceSessionIdRequest.java [hunk 2]
```diff
@@ -5,75 +5,52 @@
...
- * @zm-api-command-description Generate a new Cisco Presence server session ID and persist 
- * the newly generated session id in zimbraUCCiscoPresenceSessionId attribute for the 
- * specified UC service.
+ * @zm-api-command-description Generate a new Cisco Presence server session ID and persist the newly
+ *     generated session id in zimbraUCCiscoPresenceSessionId attribute for the specified UC
+ *     service.
  */
 @XmlAccessorType(XmlAccessType.NONE)
-@XmlRootElement(name=VoiceAdminConstants.E_UPDATE_PRESENCE_SESSION_ID_REQUEST)
+@XmlRootElement(name = VoiceAdminConstants.E_UPDATE_PRESENCE_SESSION_ID_REQUEST)
```
[the remark this answers is not in the document]
Comment (Polpetta) [reply soap/src/main/java/com/zimbra/soap/admin/message/UpdatePresenceSessionIdRequest.java]: Done!

File: store/src/main/java/com/zimbra/cs/account/ProvUtil.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Polpetta) [reply store/src/main/java/com/zimbra/cs/account/ProvUtil.java:3132]: I checked the codebase and it seemed is was related only to UC stuff in one way or another, so I removed it altogether

[the remark this answers is not in the document]
Comment (Polpetta) [reply store/src/main/java/com/zimbra/cs/account/ProvUtil.java:3132]: Looking at what it does, first it checks for domains or uc services (depending on the flag), and then it calls  `prov.countObjects(type, domain, ucService);` which iirc it didn't make too much sense without `ucservice` variable

[the remark this answers is not in the document]
Comment (keshavbhatt) [reply store/src/main/java/com/zimbra/cs/account/ProvUtil.java:3132]: NO,

**The the logic works like this:**

` countObjects {userAccount|account|alias|dl|domain|cos|server|calresource|accountOnUCService|cosOnUCService|domainOnUCService|internalUserAccount|internalArchivingAccount|internalUserAccountX} [-d {domain|id}] [-u {UCService|id}]`

So, user can either count mentioned objects in domain or UCService.

**For example:**

- to count number of userAccount in domain test.com
`zmprov countObjects userAccount -d test.com`

**We need to keep the above usecase valid and working while dropping support for UCService and its object types.**

File: soap/src/main/java/com/zimbra/soap/admin/type/CountObjectsType.java [hunk 3]
```diff
@@ -9,57 +9,45 @@
...
-        return joiner.join(CountObjectsType.values());
-    }
-    
-    public boolean allowsDomain() {
-        return allowsDomain;
-    }
-    
-    public boolean allowsUCService() {
-        return allowsUCService;
+  private CountObjectsType(boolean allowsDomain) {
```
Comment (keshavbhatt) [inline soap/src/main/java/com/zimbra/soap/admin/type/CountObjectsType.java:28]: we can drop private here, as its a enum constructor.

**Review by keshavbhatt — CHANGES_REQUESTED** [review]
We need to keep the CountObjects capability. I added more details.

File: store/src/main/java/com/zimbra/cs/account/ProvUtil.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Polpetta) [reply store/src/main/java/com/zimbra/cs/account/ProvUtil.java:3132]: Restored

File: soap/src/main/java/com/zimbra/soap/admin/type/CountObjectsType.java (hunk 3 above)
Comment (Polpetta) [reply soap/src/main/java/com/zimbra/soap/admin/type/CountObjectsType.java]: Done

**Review by keshavbhatt — APPROVED** [review]
LGTM.

**Review by frisonisland — APPROVED** [review]

**Pull request merged**